requirements.txt

```diff
@@ -1,9 +1,9 @@
 qiskit-terra>=0.17.0
 scipy>=1.4
 numpy>=1.17
-psutil>=5
 docplex; sys_platform != 'darwin'
 docplex==2.15.194; sys_platform == 'darwin'
 setuptools>=40.1.0
-retworkx>=0.7.0
+networkx>=2.2
+matplotlib
```

Comment:
I did not see any code that was added here that depends on `matplotlib` did I miss something? I get that we would prefer networkx at this application level since users know that better

Reply:
`networkx.draw` depends on matplotlib.
https://networkx.org/documentation/stable//reference/drawing.html

This line also depends on matplotlib
https://github.com/Qiskit/qiskit-optimization/blob/d0ffed7418ac01a14b96bdf3ae6ff4e211193d9e/qiskit_optimization/applications/vehicle_routing.py#L19

Reply:
matplotlib is a dependent of a dependent. If you look networkx has it as an extra_requires meaning if you do not want to use draw you not need to install it. So if a user at our level does not want to use draw we should not force it to be installed in my view if its optional in the dependent. I would not be adverse to having some extra_requires at our level - and state in the docs that to draw you need this.

Reply:
As to the vehicle_routing it seems to need it solely for some color definition `edge_cmap=plt.cm.plasma` which presumably we should be able to do without requiring matplotlib to define some colors for this purpose

Reply:
I think draw is one of the main features of the application classes. So, we need more discussion to make matplotlib extra_requires. I will make another issue for that purpose. But, I would like to merge this pr as is because matplotlib is currently required by draw.